protocol_connect.go

```diff
@@ -217,6 +220,10 @@ type connectClient struct {
 	protocolClientParams
 }
 
+func (c *connectClient) Peer() Peer {
+	return newPeerFromURL(c.URL)
```

Comment:
This appears to be the URL passed to the client constructor. Do you think people would want to have this be set to the URL which was ultimately accessed (in case of redirects)?

Reply:
Possibly? I'm not sure how to accomplish that, though, since we need this information _before_ we've attempted to make a request.